File: src/my_py_pkg/my_py_pkg/services/add_two_ints_client_no_oop.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


def main(args=None):
    rclpy.init(args=args)
    node = Node("add_two_ints_client_no_oop")
    
    # Create a client for the AddTwoInts service
    client = node.create_client(AddTwoInts, "add_two_ints")
    # Wait until the service is available
    while not client.wait_for_service(1):
        node.get_logger().warn("Waiting for add_two_ints service!")

    # Create a request and set the values to add
    request = AddTwoInts.Request()
    request.a = 11
    request.b = 373

    # Call the service asynchronously
    future = client.call_async(request)
    # Wait for the service call to complete
    rclpy.spin_until_future_complete(node, future)
    response = future.result()
    # Check if the response is valid and log the result
    if response is not None:
        node.get_logger().info(f'Result: {response.sum}')
    else:
        node.get_logger().error('Service call failed')

    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/services/add_two_ints_client_no_oop.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"
#include <sstream>

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("add_two_ints_client_no_oop"); // MODIFY NAME

    auto client = node->create_client<example_interfaces::srv::AddTwoInts> ("add_two_ints");
    while (!client->wait_for_service(std::chrono::seconds(1))){
        RCLCPP_WARN(node->get_logger(), "waiting for server...");
    }

    // Send a request
    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 2;
    request->b = 3;
    auto result = client->async_send_request(request);
    rclcpp::spin_until_future_complete(node, result);

    auto response = result.get();

    std::ostringstream output;
    output << request->a << " + " << request->b << " = " << response->sum;
    RCLCPP_INFO(node->get_logger(), "%s", output.str().c_str());

    rclcpp::shutdown();
    return 0;
}
